packages/machine/test/integration/test-io-provider.ts

```diff
@@ -133,6 +123,7 @@ export class TestIOProvider {
       r => (resolve = r)
     );
 
+    // 🤮 listen for either multisig or appId depending on message.actionName
```

Comment:
Can you comment on what's going on here

Reply:
You want me to expand this comment?

The code that runs on the IO_WAIT instruction (i.e., this function) checks the ActionName and based on that registers different types of message-received callback. It would be better to just wait for the "next message" but that involves a small bit of work.